framework/src/actions/Action.C

```diff
@@ -195,28 +201,29 @@ Action::addRelationshipManagers(Moose::RelationshipManagerType input_rm_type,
     auto & rm_type = std::get<1>(buildable_type);
     auto rm_input_parameter_func = std::get<2>(buildable_type);
 
-    // We need to be smart. If an object requested a relationship manager that is both an
-    // algebraic and coupling functor, then we need to duplicate that object. This is because we
-    // need separate objects for non-linear and auxiliary systems. The non-linear system will use a
-    // custom CouplingMatrix based on that specified in the input file for optimal sparsity. The
-    // auxiliary system will use a null CouplingMatrix (e.g. all solution vector entries on ghosted
-    // elements should be evaluable)
-
-    // check whether we need to limit the systems (and subsequent dof_maps) that this rm can be
-    // applied to
-    auto evaluable_coupleable =
-        Moose::RelationshipManagerType::COUPLING | Moose::RelationshipManagerType::ALGEBRAIC;
-    auto is_evaluable_and_coupleable = (rm_type & evaluable_coupleable) == evaluable_coupleable;
-    auto sys_type = is_evaluable_and_coupleable ? Moose::VarKindType::VAR_NONLINEAR
-                                                : Moose::VarKindType::VAR_ANY;
+    // If we have an algebraic ghosting functor, then we are going to duplicate the
+    // RelationshipManager object. This makes bookkeeping for relationship manager coverage more
+    // straightforward, i.e. when we are deciding whether to add an algebraic relationship manager,
+    // we can easily check whether a given System/DofMap is already covered or not. If we have an
+    // object that is strictly a coupling functor, we will not duplicate it and we will apply it
+    // just to the NonlinearSystem(Base). Duplicating RMs for the DofMaps is also very important for
+    // ensuring that a non-null CouplingMatrix (reflecting coupling in the nonlinear system) doesn't
+    // get applied to the auxiliary system
+
+    bool is_algebraic = (rm_type & Moose::RelationshipManagerType::ALGEBRAIC) ==
```

Comment:
We have the shortcut methods in RelationshipManager.h (`isType`, `getType`, etc)

Reply:
Those require having an instance which I do not yet have here

Reply:
static methods?

Reply:
That takes two arguments to distinguish it from the existing single argument case? I guess I can do that although I'll have to figure out a good method name to clearly distinguish which argument will used in the `&` operation and as the comparator.

Reply:
Ah I got ya...yea maybe a collection of static methods like `isAlgebraic(rm_type)`, `isGeometric(rm_type)`...I like that